unit/unit.go

```diff
@@ -169,6 +170,19 @@ func (h *Hash) Empty() bool {
 	return *h == Hash{}
 }
 
+func HashFromHexString(key string) (Hash, error) {
```

Comment:
It's not a huge deal but I would rather see it split out into an individual unit test since it's a distinct function